djoser/serializers.py

```diff
@@ -62,9 +62,13 @@ def create(self, validated_data):
         return user
 
     def perform_create(self, validated_data):
-        if config.SEND_ACTIVATION_EMAIL:
-            validated_data.update({'is_active': False})
-        return User.objects.create_user(**validated_data)
+        with transaction.atomic():
+            user = User.objects.create_user(**validated_data)
+            if config.SEND_ACTIVATION_EMAIL:
+                user.is_active = False
+                user.save(update_fields=['is_active'])
+        return user
+
```

Comment:
It could be almost like you did:

```
with transaction.atomic():
    if config.SEND_ACTIVATION_EMAIL:
        validated_data.update({'is_active': False})
    return User.objects.create_user(**validated_data)
```